src/controllers/rlglue_controller.cpp

```diff
@@ -244,8 +256,22 @@ reward_observation_terminal_t RLGlueController::constructRewardObservationTermin
   // Copy RAM and screen into our big int-vector observation 
   for (size_t i = 0; i < ram.size(); i++)
     m_observation.intArray[index++] = ram.get(i);
-  for (size_t i = 0; i < screen.arraySize(); i++)
-    m_observation.intArray[index++] = screen.getArray()[i];
+
+  if (m_osystem->settings().getBool("send_rgb")) {
```

Comment:
There's a possible buffer overrun here if the send_rgb flag is modified after creating the intArray. One solution is to store send_rgb as a const member variable and rely on this; or at least assert that the intArray size matches arraySize * 3.